src/performance/osu/OsuScore.cpp

```diff
@@ -143,8 +144,9 @@ void OsuScore::computeSpeedValue(const Beatmap& beatmap)
 					  (numTotalHits > 2000 ? log10(static_cast<f32>(numTotalHits) / 2000.0f) * 0.5f : 0.0f);
 	_speedValue *= lengthBonus;
 
-	// Penalize misses exponentially. This mainly fixes tag4 maps and the likes until a per-hitobject solution is available
-	_speedValue *= pow(0.97f, _numMiss);
+	// Penalize misses by assessing # of misses relative to the total # of objects. Default a 3% reduction for any # of misses.
+	if (_numMiss > 0)
+		_speedValue *= 0.97f * std::pow(1.0f - std::pow(_numMiss / static_cast<f32>(numTotalHits), 0.775f), std::pow(static_cast<f32>(_numMiss), 0.75f));
```

Comment:
Just so that this doesn't slip by, this was changed to

```suggestion
		_speedValue *= 0.97f * std::pow(1.0f - std::pow(_numMiss / static_cast<f32>(numTotalHits), 0.775f), std::pow(static_cast<f32>(_numMiss), 0.875f));
```

lazer-side in https://github.com/ppy/osu/pull/11147/commits/b7872a54b885a2572ec439e8e2a8c0cb2f47dccf.